flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -2838,6 +2843,37 @@ mlir::Value IntrinsicLibrary::genSign(mlir::Type resultType,
   return builder.create<mlir::SelectOp>(loc, cmp, neg, abs);
 }
 
+// SIZE
+fir::ExtendedValue
+IntrinsicLibrary::genSize(mlir::Type resultType,
+                          llvm::ArrayRef<fir::ExtendedValue> args) {
+  // TODO: handle assumed-rank arrays, especially a dummy whose actual argument
+  // is an assumed-size array
+  assert(args.size() == 3);
+
+  // Calls to SIZE that don't have the DIM argument are handled elsewhere
+  assert(!isAbsent(args[1]));
+
+  // Handle the ARRAY argument
+  auto array = builder.createBox(loc, args[0]);
+
+  // Handle the optional DIM argument.  Convert the Fortran 1-based index to
+  // the FIR 0-based index
+  auto indexType = builder.getIndexType();
+  auto oneBasedDim =
+      builder.createConvert(loc, indexType, fir::getBase(args[1]));
+  auto one = builder.createIntegerConstant(loc, indexType, 1);
```

Comment:
The comemnt about says that the DIM argument is optional. But I only see a reference to args[1]. Is there something happening under the hood when args[1] is an absent argument? If so, can you add a comment that explains that. I bring this up because most, if not all, other intrinsics in this file have special handling when an optional argument is not present.

Reply:
Will do.